### PR TITLE
rofi-emoji: fix missing dependencies `xdotools` `wtype`

### PR DESCRIPTION
**Pull request by juliendiot42**
Dear @cole-h, thank you for maintaining `rofi-emoji`.
\cc @Mange 

I have recently installed this package but found a bug. When trying to insert an emoji, and error notification raises:
```
Could not find any tool to handle insertion. Please install xdotool or wtype.
```

I suggest this merge request to fix this bug.
Please let me know if further changes are required.

Best regards,
@juliendiot42

## Description of changes

This commit add those missing dependencies.
To control which of these tools should be installed (one is for x11 and the other for wayland) 2 derivations arguments have been created: `x11Support` and `waylandSupport` defaulting to `true`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)

> Tested on **`x11` only**.
> After `nix build .\#rofi-emoji`
> Previous behaviour: 
>    ```sh
>    echo "the text" | ./result/share/rofi-emoji/clipboard-adapter.sh insert
>    ```
> returns
>    ```
>    Could not find any tool to handle insertion. Please install xdotool or wtype.
>    ```
> Fixed behaviour:
>    ```sh
>    echo "the text" | ./result/share/rofi-emoji/clipboard-adapter.sh insert
>    ```
> returns
>    ```
>    the text
>    ```
> And `the text` is inserted in the terminal.
> When tested with rofi it inserts the emoji without the notification error.

- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [?] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).
> I read the guidelines but this is my first contribution, I am not sure the code is formatted correctly, I am sorry. 

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/misc/rofi-emoji/default.nix [hunk 1]
```diff
@@ -55,6 +63,8 @@ stdenv.mkDerivation rec {
     wl-clipboard
     xclip
     xsel
+    xdotool
+    wtype
```
Comment (cole-h) [inline pkgs/applications/misc/rofi-emoji/default.nix:67]: Do these need to be in the `nativeBuildInputs` too, and not just added to the `PATH` as you did above? (If so, it'd be nice to guard them on the `waylandSupport` and `x11Support` flags, too.)

Comment (juliendiot42) [reply pkgs/applications/misc/rofi-emoji/default.nix]: Indeed, those are only runtime dependencies I guess.  
I removed them from `nativeBuildInputs`.

**Review by cole-h — APPROVED** [review]
Thanks!

(In a future PR we may want to move `wl-clipboard` and `xclip` + `xsel` to the `*Support` optionals but that's not blocking)

**Comment by Mange** [comment]
Thanks!

**Pull request merged**